src/include/sof/math/exp_fcn_hifi.h

```diff
@@ -0,0 +1,45 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOFM_EXP_HIFI_H__
+#define __SOFM_EXP_HIFI_H__
+
+#include <sof/platform.h>
+#include <stdint.h>
+#include <stddef.h>
+
+#undef EXPONENTIAL_GENERIC
+#undef EXPONENTIAL_HIFI3
+#undef EXPONENTIAL_HIFI4
+#undef EXPONENTIAL_HIFI5
```

Comment:
What is the purpose of all these `#undef`s? To keep things simple macros should have a single source of truth. They should not be defined in different places fighting with each other and `#undef`-ining each other.